waspc/data/Generator/templates/react-app/src/queryClient.js

```diff
@@ -1,3 +1,16 @@
 import { QueryClient } from 'react-query'
 
-export const queryClient = new QueryClient()
+let resolveQueryClientInitialized, isQueryClientSetUp
+export const queryClientInitialized = new Promise(resolve => {
+  resolveQueryClientInitialized = resolve
+});
```

Comment:
This pattern is courtesy of Shaynei, I stole it from the pg-boss PR. Thanks @shayneczyzewski!